central/sac/root_scope_checker_core_context_enricher.go

```diff
@@ -37,73 +32,30 @@ func GetEnricher() *Enricher {
-// scope checking. There are four possibilities currently:
-//   1. User identity maps to a special case: (a) deny all or (b) allow all =>
-//      add the corresponding trivial scope checker.
-//   2. Built-in scoped authorizer must be used => use the scope checker
-//      constructed from resolved roles associated with the identity.
-//   3. Auth plugin is detected => use the scope checker created around the
-//      auth plugin client in use at the time of request.
-//   4. Unrecoverable error => nil context.
+// scope checking.
 func (se *Enricher) getRootScopeCheckerCore(ctx context.Context) (context.Context, observe.ScopeCheckerCoreType, error) {
-	client := se.clientManager.GetClient()
-
 	// Check the id of the context and decide scope checker to use.
 	id := authn.IdentityFromContextOrNil(ctx)
 	if id == nil {
-		// 1a. User identity not found => deny all.
+		// User identity not found => deny all.
 		return sac.WithGlobalAccessScopeChecker(ctx, sac.DenyAllAccessScopeChecker()), observe.ScopeCheckerDenyForNoID, nil
 	}
 	if id.Service() != nil || userpass.IsLocalAdmin(id) {
-		// 1b. Admin => allow all.
+		// Admin => allow all.
 		return sac.WithGlobalAccessScopeChecker(ctx, sac.AllowAllAccessScopeChecker()), observe.ScopeCheckerAllowAdminAndService, nil
 	}
 	if len(id.Roles()) == 0 {
-		// 1c. User has no valid role => deny all.
+		// User has no valid role => deny all.
 		return sac.WithGlobalAccessScopeChecker(ctx, sac.DenyAllAccessScopeChecker()), observe.ScopeCheckerDenyForNoID, nil
 	}
-	if client == nil {
-		// 2. Built-in scoped authorizer must be used.
-		scopeChecker, err := authorizer.NewBuiltInScopeChecker(ctx, id.Roles())
-		if err != nil {
-			return nil, observe.ScopeCheckerNone, errors.Wrap(err, "creating scoped authorizer for identity")
-		}
-		return sac.WithGlobalAccessScopeChecker(ctx, scopeChecker), observe.ScopeCheckerBuiltIn, nil
-	}
-	ctx = sac.SetContextPluginScopedAuthzEnabled(ctx)
-
-	// Get the principal and the cache key for it.
-	principal, idCacheKey, err := idToPrincipalAndCacheKey(id)
+	scopeChecker, err := authorizer.NewBuiltInScopeChecker(ctx, id.Roles())
 	if err != nil {
-		return nil, observe.ScopeCheckerNone, err
-	}
-
-	// 3. If we have a scope checker cached for the user, use that,
-	// otherwise generate a new one and add it to the cache.
-	cacheForClient := se.cacheForClient(client)
-	rsc, _ := cacheForClient.Get(idCacheKey).(sac.ScopeCheckerCore)
-	if rsc == nil {
-		rsc = sac.NewRootScopeCheckerCore(NewRequestTracker(client, datastore.Singleton(), principal))
-		// Not locking here can cause multiple root contexts to be created for
-		// one user. This will have correct results and be eventually consistent
-		// but it will be slightly inefficient.
-		cacheForClient.Add(idCacheKey, rsc)
+		return nil, observe.ScopeCheckerNone, errors.Wrap(err, "creating scoped authorizer for identity")
 	}
-	return sac.WithGlobalAccessScopeChecker(ctx, rsc), observe.ScopeCheckerPlugin, nil
```

Comment:
The constant `ScopeCheckerPlugin` in `pkg/sac/observe/authz_trace.go` may be removed.